apps/web/src/stores/editor-store.ts

```diff
@@ -871,12 +871,20 @@ class EditorStore extends BaseStore<EditorStore> {
   };
 
   newSession = () => {
-    this.addSession({
-      type: "new",
-      id: getId(),
-      context: useNoteStore.getState().context,
-      saveState: SaveState.NotSaved
-    });
+    const state = useEditorStore.getState();
+    if (state.sessions.some((session) => session.type === "new")) {
+      const invalidSession = state.sessions.find(
+        (session) => session.type === "new"
+      );
```

Comment:
No need to loop twice over the sessions. Move the `.find` out and just check its return value in the if statement.

Reply:
Also can we not call it `invalidSession`? There's nothing invalid about it.

Reply:
Like this?

```
  newSession = () => {
    const state = useEditorStore.getState();
    const newSession = state.sessions.find((session) => session.type === "new");
    if (newSession) {
      this.activateSession(newSession?.id);
    } else {
      this.addSession({
        type: "new",
        id: getId(),
        context: useNoteStore.getState().context,
        saveState: SaveState.NotSaved
      });
    }
  };
```

Reply:
Like this?

```
  newSession = () => {
    const state = useEditorStore.getState();
    const session = state.sessions.find((session) => session.type === "new");
    if (session) {
      this.activateSession(session?.id);
    } else {
      this.addSession({
        type: "new",
        id: getId(),
        context: useNoteStore.getState().context,
        saveState: SaveState.NotSaved
      });
    }
  };
```

Reply:
Yes